Kickstarter-iOS/Views/Cells/SettingsTextInputCell.swift

```diff
@@ -0,0 +1,121 @@
+import Prelude
+import UIKit
+
+private struct Margin {
+  static let leftRight: CGFloat = 8
+  static let topBottom: CGFloat = 15
+}
+
+class SettingsTextInputCell: UITableViewCell {
+  // MARK: - Accessors
+
+  private lazy var stackView: UIStackView = {
+    let layoutMargins = UIEdgeInsets.init(topBottom: Margin.topBottom, leftRight: Margin.leftRight)
+    return UIStackView()
+      |> \.isLayoutMarginsRelativeArrangement .~ true
+      |> \.layoutMargins .~ layoutMargins
+      |> \.spacing .~ 8
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
+
+  private lazy var label: UILabel = {
+    return UILabel()
+      |> \.adjustsFontForContentSizeCategory .~ true
+      |> \.font .~ UIFont.preferredFont(forTextStyle: .body)
```

Comment:
Our settings styles don't use `UIFont` properly in order to support Dynamic Type. Therefore I've opted for basic implementation (the could potentially be extracted to a style, but I didn't see the value at this step)

Reply:
I'm also curious, why do you say our styles don't support dynamic type? All of our font functions (`.ksr_body()` etc) under the hood use `UIFont.preferredFont(forTextStyle: .body)`

Reply:
I was able to make the above work (resize the font when the font size changes) with the following:

```
|> \.adjustsFontForContentSizeCategory .~ true
|> \.font .~ .ksr_body()
```

so it looks like `adjustsFontForContentSizeCategory` is the key for this to properly resize the `UILabel`/`UITextField` on iOS 10+.

I originally suspected the problem was in `Fonts.swift` where we declare a convenience font accessor which calculates font size based on hard coded values

```
static func preferredFont(style: UIFont.TextStyle, size: CGFloat? = nil) -> UIFont { }
```

and is used by majority of our styles (`.ksr_body()`, etc.). If you look at the font that's being returned from the method it's actually constructed using `descriptor` not `preferredFont` so my initial assumption was that this was causing Dynamic Type to not resize properly, but it seems like that's not the case. This method basically uses custom scale so that we can tweak the size of the font to our likings.

@Scollaco do you have (by any chance) any context to why we implemented `preferredFont` this way? Looking at the method the two sole contributors is Brandon and you.

Reply:
Thanks for digging in. Unfortunately this was implemented way before I started working here. Maybe @justinswart has a little context?

Reply:
I noticed you've mostly just updated it for Swift 4.2 so wanted to be sure you're not holding some secrets Nino :) 

For clarification, I've chatted with Justin offline and it appears we're using this constructor in order to customize the font scale. You can easily replace the implementation of this with 
```
fileprivate static func preferredFont(style: UIFont.TextStyle, size: CGFloat? = nil) -> UIFont {
    return UIFont.preferredFont(forTextStyle: style)
}
```
and see the font difference when running snapshot tests.